src/components/engagement/engagement.service.ts

```diff
@@ -162,6 +163,78 @@ export class EngagementService {
     return result ? true : false;
   }
 
+  /**
+   * if firstScripture is true, validate that the engagement
+   * is the only engagement for the language that has firstScripture=true
+   * that the language doesn't have hasExternalFirstScripture=true
+   */
+  protected async verifyFirstScripture(
+    engagementId?: string,
+    languageId?: string
```

Comment:
Is one of these always required?

If so,
```tsx
verifyFirstScripture({ engagementId, languageId }: MergeExclusive<{ engagementId: string }, { languageId: string }>) {
```
Would be a better way to define it as it enforces that in TS and doesn't look awkward for one of the cases:
```tsx
await this.verifyFirstScripture({ languageId });
await this.verifyFirstScripture({ engagementId });
```